ldap/servers/slapd/back-ldbm/db-bdb/bdb_misc.c

```diff
@@ -314,37 +291,100 @@ bdb_start_autotune(struct ldbminfo *li)
     }
     total_cache_size += li->li_dbcachesize;
 
-    /* For each backend */
-    /*   apply the appropriate cache size if 0 */
-    if (backend_count > 0) {
-        li->li_cache_autosize_ec = entry_size;
-        li->li_dncache_autosize_ec = dn_size;
-    }
-
     for (inst_obj = objset_first_obj(li->li_instance_set); inst_obj;
-         inst_obj = objset_next_obj(li->li_instance_set, inst_obj)) {
-
+         inst_obj = objset_next_obj(li->li_instance_set, inst_obj))
+    {
         inst = (ldbm_instance *)object_get_data(inst_obj);
         uint64_t cache_size = (uint64_t)cache_get_max_size(&(inst->inst_cache));
         uint64_t dncache_size = (uint64_t)cache_get_max_size(&(inst->inst_dncache));
+        uint64_t ec_size = MINCACHESIZE;
+        uint64_t dn_size = DEFAULT_DNCACHE_SIZE;
+        uint64_t clamp_div = 0; /* For clamping the autotune value to a 64Mb boundary */
+
+        /* Calculate this instance's slice of memory it can use */
+        if (total_pages) {
```

Comment:
Do I understand correctly that `total_pages == 0` when DB doesn't have any entries?

So if we create a DB with no entries and then start to add them, we might end up in a situation when the cache is incorrectly auto tuned.
I wonder if should add some kind of warning when `total_pages == 0` (i.e. "No pages found in any backend, using minimum cache sizes"). Or something like that.

I understand it might be an edge case which will be resolved on restart, but still I wonder.